src/ansys/fluent/core/quantity.py

```diff
@@ -0,0 +1,601 @@
+                unit_str, term_power = term[: term.index("^")], float(
+                    term[term.index("^") + 1 :]
+                )
+                has_multiplier = not (
+                    unit_str in _UnitsTable.fundamental_units
+                    or unit_str in _UnitsTable.derived_units
+                )
+                unit_str = remove_multiplier(unit_str) if has_multiplier else unit_str
+
+            term_power *= power
+            unit_dim = self._get_dim(unit_str, term_power)
+
+            if unit_dim != None:
+                self._add_or_update_dimension(unit_dim)
+
+    def as_dict(self):
+        return self._dimensions
+
+    def _get_dim(self, unit_str, power):
+        if unit_str in _UnitsTable.fundamental_units:
+            return {_UnitsTable.fundamental_units[unit_str]: power}
+        elif unit_str in _UnitsTable.derived_units:
+            self._parser(_UnitsTable.derived_units[unit_str], power)
+        elif unit_str in _UnitsTable.derived_units_with_conversion_factor:
+            _, unit_str = _UnitsTable.derived_units_with_conversion_factor[unit_str]
+            self._parser(unit_str, power)
+        else:
+            raise ValueError("Not implemented")
+
+
+def get_si_unit_from_dim(dim_list):
+    si_unit = ""
+    dim_to_unit_map = {
+        "M": "kg",
+        "L": "m",
+        "T": "s",
+        "Temp": "K",
+        "I": "A",
+        "N": "mol",
+        "J": "cd",
+        "Angle": "radian",
+        "SAngle": "sr",
+        "": "",
+    }
+    for key, power in zip(_UnitsTable.dimension_order.values(), dim_list):
+        unit_str = dim_to_unit_map[key]
+        spacechar = " " if len(si_unit) > 0 else ""
+        if power > 0.0 or power < 0.0:
+            si_unit += spacechar + unit_str + "^" + str(power)
+        elif power == 1.0:
+            si_unit += spacechar + unit_str
+    return si_unit
+
+
+class UnitSystem:
+
+    _dim_to_unit_sys_map = {
+        "M": {"SI": "kg", "CGS": "g", "BTU": "slug"},
+        "L": {"SI": "m", "CGS": "cm", "BTU": "ft"},
+        "T": {"SI": "s", "CGS": "s", "BTU": "s"},
+        "Temp": {"SI": "K", "CGS": "K", "BTU": "R"},
+        "I": {"SI": "A", "CGS": "A", "BTU": "A"},
+        "N": {"SI": "mol", "CGS": "mol", "BTU": "slugmol"},
+        "J": {"SI": "cd", "CGS": "cd", "BTU": "cd"},
+        "Angle": {"SI": "radian", "CGS": "radian", "BTU": "radian"},
+        "SAngle": {"SI": "sr", "CGS": "sr", "BTU": "sr"},
+        "": {"SI": "", "CGS": "", "BTU": ""},
+    }
+
+    _supported_unit_sys = ("SI", "CGS", "BTU")
+
+    def __init__(self, unit_sys):
+        self._unit_system = unit_sys.upper()
+
+        if self._unit_system not in UnitSystem._supported_unit_sys:
+            raise ValueError(
+                "Unsupported unit system, only 'SI', 'CGS', 'BTU' is allowed."
+            )
+
+    def _get_unit_from_dim(self, dim_list):
+        unit = ""
+
+        for key, power in zip(_UnitsTable.dimension_order.values(), dim_list):
```

Comment:
if you make the change above then I think this can be:
```
base_units = UnitSystem._dim_to_unit_sys_map[self._unit_system]
for key, power in zip(_UnitsTable.dimension_order.values(), dim_list):
    unit_str = base_units[key]
```

```

Reply:
`elif power == 1` is unreachable. Is this tested?

Reply:
``` python
if power == 1:
  unit += spacechar + base_units[key]
elif power != 0:
  unit += spacechar + base_units[key] + "^" + str(int(power) if power.is_integer() else power)
```

Reply:
Thank you very much @dnwillia-work , @h-krishnan .